### PR TITLE
Replacing PyMC3 plots w/ Arviz plots & sigma Param change [Part 3]

### PR DESCRIPTION
**Pull request by CloudChaoszero**
# Description
The following is a large PR breakdown of PR #16.

Replace PyMC3 dependent plots with arviz plots in case studies & examples.
* The issue is a 3-part step for the following:
  * Initiative via Issue https://github.com/pymc-devs/pymc3/issues/4371 (Leave open)

Replace parameter `sd` with `sigma` (e.g. some examples have `pm.Normal(...sd=...)`
* Closes issue #2

### CODE REVIEW

**Comment by MarcoGorelli** [comment]
Hi @CloudChaoszero - it looks like the watermarks haven't changed, could you please run the notebooks top-to-bottom? It may be that some other parts need updating (I've often found that when updating old notebooks)

**Comment by CloudChaoszero** [comment]
@MarcoGorelli Pardon about that. haha I went ahead and did the full runs, and squashed the commits.

**Comment by CloudChaoszero** [comment]
# Small note
There were a couple of errors I encountered like:

## examples/diagnostics_and_criticism/Diagnosing_biased_Inference_with_Divergences.ipynb:
* `divergent_point` had a dimension/indice error in section
```python
ii = 5

tau_log_d = divergent_point["tau_log__"]
theta0_d = divergent_point["theta"][:, ii]
Ndiv_recorded = len(tau_log_d)
```
So I removed the `[:, ii]`, since it's a singular dimension array, and not 2-dim.

## examples/diagnostics_and_criticism/Bayes_factor.ipynb:
* parallel=True caused an error. It is recorded as a defect, based on [this issue](https://github.com/pymc-devs/pymc3/issues/4324) 
```
        trace = pm.sample_smc(1000, random_seed=42, parallel=True)

```

**Comment by MarcoGorelli** [comment]
Thanks @CloudChaoszero for updating!

TBH I'm really not keen on adding
```python
warnings.filterwarnings("ignore", category=FutureWarning)
warnings.filterwarnings("ignore", category=UserWarning)
```
- if there's a specific warning that needs disabling in one cell, then OK, but disabling everywhere doesn't strike me as a great practice. CC @AlexAndorra any thoughts on disabling warnings in notebooks?

**Comment by AlexAndorra** [comment]
> If there's a specific warning that needs disabling in one cell, then OK, but disabling everywhere doesn't strike me as a great practice. CC @AlexAndorra any thoughts on disabling warnings in notebooks?

I agree with @MarcoGorelli : we already know many people don't pay enough attention to warnings, so if we also disable them in notebooks, that makes our job more difficult. On the contrary, displaying them in notebooks is a great way to make people more aware of important warnings -- for instance, users should really know that we're gonna switch to `InferenceData` by default at one point, and make breaking changes in the next major version, etc.

File: examples/diagnostics_and_criticism/model_averaging.ipynb [hunk 1]
```diff
@@ -2,14 +2,8 @@
  "cells": [
```
Comment (OriolAbril) [inline examples/diagnostics_and_criticism/model_averaging.ipynb:2]: <p>I suggest using:</p><p> </p><pre class="ql-syntax" spellcheck="false">ppc_w = pm.sample_posterior_predictive_w(
&nbsp;&nbsp;&nbsp; traces,
&nbsp;&nbsp;&nbsp; models=[model_0, model_1, model_2],
&nbsp;&nbsp;&nbsp; weights=comp.weight.sort_index(ascending=True),
&nbsp;&nbsp;&nbsp; progressbar=True,
)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/25/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

Comment (OriolAbril) [inline examples/diagnostics_and_criticism/model_averaging.ipynb:2]: <p>and here:</p><p> </p><pre class="ql-syntax" spellcheck="false">ppc_2 = pm.sample_posterior_predictive(trace_2, model=model_2, progressbar=False)
</pre>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/25/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

Comment (OriolAbril) [inline examples/diagnostics_and_criticism/model_averaging.ipynb:2]: <p>Not sure why there is a mean after the call to hdi, I would suggest doing:</p><p> </p><pre class="ql-syntax" spellcheck="false">hpd_w = az.hdi(ppc_w["kcal"].flatten())
</pre><p> </p><p>instead. If I'm not wrong, between this and the changes to <code>sample_...</code>  there won't be a need to ignore warnings anymore everything will have been solved and will follow best practices</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/25/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='22'/>

File: examples/diagnostics_and_criticism/model_comparison.ipynb [hunk 2]
```diff
@@ -2,26 +2,48 @@
  "cells": [
```
Comment (OriolAbril) [inline examples/diagnostics_and_criticism/model_comparison.ipynb:2]: <p>There should be no need to ignore warnings in this notebook, ArviZ does not warn about the change in default anymore.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/25/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

File: examples/diagnostics_and_criticism/sampler-stats.ipynb [hunk 3]
```diff
@@ -16,15 +16,36 @@
    "cell_type": "code",
```
Comment (OriolAbril) [inline examples/diagnostics_and_criticism/sampler-stats.ipynb:16]: <p>This plot is actually <code>az.plot_energy</code> the cell can be reduced to a single line (could be done in a follow up PR)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/pymc-devs/pymc-examples/pull/25/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='18'/>

**Comment by CloudChaoszero** [comment]
> > If there's a specific warning that needs disabling in one cell, then OK, but disabling everywhere doesn't strike me as a great practice. CC @AlexAndorra any thoughts on disabling warnings in notebooks?
> 
> I agree with @MarcoGorelli : we already know many people don't pay enough attention to warnings, so if we also disable them in notebooks, that makes our job more difficult. On the contrary, displaying them in notebooks is a great way to make people more aware of important warnings -- for instance, users should really know that we're gonna switch to `InferenceData` by default at one point, and make breaking changes in the next major version, etc.

Yeah, I just did it to not show my local path when the Warnings come up. lol

I can remove them going forward, if you all are okay with that.

**Review by OriolAbril — APPROVED** [review]
We should open an issue about the wrong plot parallel in the divergences notebook so we don't forget to fix eventually everything else looks good

**Pull request merged**